enterprise/cloud.amazon/src/org/netbeans/modules/cloud/amazon/AmazonInstance.java

```diff
@@ -131,10 +144,10 @@ private static AWSCredentials getCredentials(String keyId, String key) {
     
     public void testConnection() {
         assert !SwingUtilities.isEventDispatchThread();
-        AWSElasticBeanstalk client = new AWSElasticBeanstalkClient(getCredentials());
-        if (regionUrl != null) {
-            client.setEndpoint(regionUrl);
-        }
+        AWSElasticBeanstalk client = AWSElasticBeanstalkClientBuilder.standard()
+                                        .withCredentials(new AWSStaticCredentialsProvider(getCredentials()))
+                                        .withRegion(regionCode)
+                                        .build();
```

Comment:
I got the expression, that this sequence is repeated in this file. Would it make sense to introduce a private method "createAWSElasticBeanstalkClient" and unify the initialization code for the client there?

Reply:
Done :)  There are a couple of methods where the credentials are being passed in from outside, for those two methods, the Client is still being constructed in those methods...